crates/extensions/c8y_mapper_ext/src/log_upload.rs

```diff
@@ -130,42 +140,41 @@ impl CumulocityConverter {
                 };
                 let event_response_id = self.http_proxy.send_event(create_event).await?;
 
-                // Send a request to the Uploader to upload the file asynchronously.
-                let uploaded_file_path = self
-                    .config
-                    .data_dir
-                    .file_transfer_dir()
-                    .join(target.external_id.as_ref())
-                    .join("log_upload")
-                    .join(format!("{}-{}", response.log_type, cmd_id));
-
-                let binary_upload_event_url = self
-                    .c8y_endpoint
-                    .get_url_for_event_binary_upload_unchecked(&event_response_id);
-
-                let upload_request = UploadRequest::new(
-                    self.auth_proxy
-                        .proxy_url(binary_upload_event_url.clone())
-                        .as_str(),
-                    &uploaded_file_path,
-                )
-                .with_content_type(ContentType::TextPlain);
+                // Send a request to the Downloader to download the file asynchronously from FTS
+                let log_filename = format!("{}-{}", response.log_type, cmd_id);
 
-                self.uploader_sender
-                    .send((cmd_id.into(), upload_request))
-                    .await
-                    .map_err(CumulocityMapperError::ChannelError)?;
+                let tedge_file_url = format!(
+                    "http://{}/tedge/file-transfer/{external_id}/log_upload/{log_filename}",
+                    &self.config.tedge_http_host,
+                    external_id = target.external_id.as_ref()
+                );
+
+                let destination_dir = tempfile::tempdir_in(&self.config.tmp_dir)
+                    .context("Failed to create a temporary directory")?;
+                let destination_path = destination_dir.path().join(log_filename);
 
-                self.pending_upload_operations.insert(
+                self.pending_fts_download_operations.insert(
                     cmd_id.into(),
-                    UploadOperationData {
+                    FtsDownloadOperationData {
+                        download_type: FtsDownloadOperationType::LogDownload,
+                        url: tedge_file_url.clone(),
+                        destination_dir,
+
                         smartrest_topic,
                         clear_cmd_topic: message.topic.clone(),
-                        c8y_binary_url: binary_upload_event_url.to_string(),
-                        operation: CumulocitySupportedOperations::C8yLogFileRequest,
+                        event_response_id,
```

Comment:
There's no point creating/sending the event before this download. So, it's better to move that to `handle_fts_log_download` right before the `binary_upload_event_url` is created.

Reply:
fixed in 5e1124f0.